src/uu/pidwait/src/wait.rs

```diff
@@ -0,0 +1,47 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use uu_pgrep::process::ProcessInformation;
+
+// Dirty, but it works.
+// TODO: Use better implementation instead
+#[cfg(target_os = "linux")]
+pub(crate) fn waiting(procs: &[ProcessInformation]) {
+    let mut list = procs.to_vec();
+
+    loop {
+        for proc in &list.clone() {
+            // Check is running
+            if !is_running(proc.pid) {
+                list.retain(|it| it.pid != proc.pid)
+            }
+        }
+
+        if list.is_empty() {
+            return;
+        }
+    }
```

Comment:
I would add some `sleep` to this loop, otherwise it uses almost 100% of a CPU ;-)